tests/test_scheduler.py

```diff
@@ -364,3 +365,20 @@ def test_custom_lock_timeout(self):
         scheduler = RedBeatScheduler(app=self.app)
         assert self.config_dict['beat_max_loop_interval'] == scheduler.max_interval
         assert self.config_dict['redbeat_lock_timeout'] == scheduler.lock_timeout
+
+
+class ReadBeatRedisConnectionFailures(RedBeatCase):
+    def test_when_scheduler_fails_to_acquire_lock_on_init_no_due_tasks_sent(self):
+        with patch.object(self.app.redbeat_redis, 'lock') as lock_mock:
+            lock_mock.return_value.acquire.side_effect = Exception(
+                "Redis lock acquire failure"
+            )
+            self.create_entry(name='next', s=due_now).save()
+            service = Service(self.app, scheduler_cls=RedBeatScheduler)
+            with patch.object(service.scheduler, 'maybe_due') as maybe_due_wrapper:
+                maybe_due_wrapper.side_effect = AssertionError(
```

Comment:
It was rather tricky to write this test. I want to test using the celery beat service itself as that is where part of the chain of bugs lies. However `start` contains an infinite while loop so we need to figure out some way of crashing out of that loop during the test. There is a lot of broad exception catching and not reraising going on so patching `maybe_due` was my best solution to actually crash the `start` method call.